src/Misc/screenshotTools.ts

```diff
@@ -126,12 +126,18 @@ export class ScreenshotTools {
 
         var scene = camera.getScene();
         var previousCamera: Nullable<Camera> = null;
+        let previousActiveCameras: Camera[] = [];
 
         if (scene.activeCamera !== camera) {
             previousCamera = scene.activeCamera;
             scene.activeCamera = camera;
         }

```

Comment:
Doesn't this already handle multi-camera setup?
are we sure the bug is not somewhere else? 
maybe scene.activeCamera is not being respected if there are multiple activeCameras?

Reply:
Well, that is a possibility since I haven't looked at it (it is far beyond my knowledge on the BabylonJS core). Also, if some method depends on that behaviour, it could bring some backward incompatibilities.